x-pack/plugins/security_solution/public/cases/pages/index.tsx

```diff
@@ -82,6 +82,10 @@ const CaseContainerComponent: React.FC = () => {
             },
           },
         });
+        telemetry.reportDetailsFlyoutOpened({
+          tableId: TimelineId.casePage,
+          panel: 'right',
+        });
```

Comment:
Would it make sense to integrate the telemetry report within the `openFlyout` (probably `openRightPanel` and `openLeftPanel` too) function? We already have the `tableId` (`scopeId`) and `panel` parameters. This way have telemetry automatically everywhere, instead of having to implement it explicitly next to all `openFlyout` calls.

Reply:
@semd one nuance is that the `open` functions are used for all kinds of navigations. For example, clicking `Expand details` and `entities` within insights both call `openLeftPanel`. 

For analytics we want to separate them as 2 types of events, first answers `how many users are aware of the expanded details`, and the second may answer `how many users are interested in entities`. Currently we don't have a way to track where the function is called.
![image](https://github.com/elastic/kibana/assets/18648970/5e991ea3-3a39-4843-ae20-6cbf64cb5a41)

We could add telemetry at the package level in a separate PR, but it will be more general usage and covers every team that uses it. (cc @PhilippeOberti )

Updated the title and description as well, we often use `document details` and `expandable flyout` interchangeably, but `Document details` is flyout in alerts table, `expandable flyout` is the package`. Hope this helps!

Reply:
I see `openLeftPanel` and `openRightPanel` functions have the `id`, which could be used to know that, isn't it? (those IDs could have better naming though)
 
I understand this implementation sends telemetry data only to answer "document details" questions. But, if we plan to have telemetry for more use cases of the flyout in the future, I think we should probably re-assess the approach here to build something more generic. If we collect telemetry from within the package, sending the correct data, we will need to implement it only once, and we'll be able to answer any question.

I am not saying the current approach is wrong, but it forces us to implement the telemetry explicitly for every use case we want to track. All those telemetry calls happen next to the `openFlyout` or `open*Panel` calls, which already receive all the information we need. It feels like they want to be integrated inside the flyout 😄

What do you think @PhilippeOberti?

Reply:
while I see why @christineweng  did it this way, I get your point @semd! Let me think about it over the weekend and @christineweng we can brainstorm on it Monday during our UIUX meeting!

Reply:
over the last couple of days I kept going back and forth on which approach would be best here, but I'm now leaning on keeping what @christineweng has, for a couple of reasons:
- it perfectly answers our needs right now, which are to understand the usage of the document details flyout. While having things in the package would be nice because it would benefit others, at this moment we do not care about that... (I do not think that in a mean way, just stating the facts)
- the complexity of the code is really low. If we were to move some of this logic in the package, we would want to have an api and interface well defined and documented. This could easily lead to us being too restrictive or too complex (or both), as we do not know what other developers might want to send to the telemetry for their flyouts. There might be a way to make things extremely generic, but usually that means the code is harder to understand and the documentation is harder to read...
- this implementation should not prevent us from moving the logic in the package later, once we know a bit more how other people use the package